RELEASING.md

```diff
@@ -1,5 +1,58 @@
 # Releasing
 
-The process for how to release the crates in this repository are documented here:
+To release Soroban RPC, follow this process **in order**:
 
-https://github.com/stellar/actions/blob/main/README-rust-release.md
+## Update Horizon Version Dependency
+1. Run `go get github.com/stellar/go@<horizon_release_tag>` to update `go.mod` and `go.sum` to match the tagged Horizon version branch.
+2. Commit this change to `main` in a pull request
+
+## Follow Rust Workflow
+Currently, the `soroban-rpc` repo doesn't publish crates, so we will be running a *modified* version of our main [Rust release workflow](https://github.com/stellar/actions/blob/main/README-rust-release.md).
+
+1. Run the [Bump Version workflow](https://github.com/stellar/actions/blob/main/README-rust-release.md#step-1-run-the-bump-version-workflow) from `main`
+2. [Review the PR](https://github.com/stellar/actions/blob/main/README-rust-release.md#2-review-pr) from running the Bump Version workflow
+3. [Merge the PR](https://github.com/stellar/actions/blob/main/README-rust-release.md#3-merge-pr) after the publish dry runs complete successfully. Although currently we don't publish crates from the `soroban-rpc` repo, we want the dry runs to succeed.
+
+## Create a GitHub Release From a Tag
+1. Create an annotated tag with `git tag -a v<release_version_number> -m "Description for release"`
+2. Push the tag to remote with `git push origin --tags`
+3. Create a new [GitHub release](https://github.com/stellar/soroban-rpc/releases/new) from the previously created tag.
+    * The release title MUST NOT start with a v, otherwise artifact uploads fail (see [workflow file](https://github.com/stellar/soroban-tools/blob/main/.github/workflows/publish.yml) and this [Slack thread](https://stellarfoundation.slack.com/archives/C04ECVCV162/p1694729751569919) for context)
+4. Monitor GitHub actions until they succeed
+
+## Push New RPC Debian Package
+
+1. Run the [Buildmeister soroban-package-builder pipeline](https://buildmeister-v3.stellar-ops.com/job/Platform/job/soroban-package-builder/)
+    * Set `VERSION_OVERRIDE` if necessary, replacing "-" with "~" (e.g. 20.0.0~rc1)
+    * Set `BRANCH` to the release tag (e.g. v20.0.0-rc1)
+    * Enable `PROMOTE_TO_TESTING`
+    * ![Pipeline Parameters](docs/img/soroban-package-builder.png)
+
+2. Review the package stage log, and note the build number for next steps
+
+![Soroban Package Logs](docs/img/soroban-package-logs.png)
+
+## Push New Docker Image
```

Comment:
There is also a promote-to-stable jenkins job that needs to be run after the apt and docker image are generated. Probably want a new section for that